lib/ui/hash_codes.dart

```diff
@@ -2,122 +2,33 @@
-    result = _Jenkins.combine(result, arg03);
-    if (!identical(arg04, _hashEnd)) {
-      result = _Jenkins.combine(result, arg04);
-      if (!identical(arg05, _hashEnd)) {
-        result = _Jenkins.combine(result, arg05);
-        if (!identical(arg06, _hashEnd)) {
-          result = _Jenkins.combine(result, arg06);
-          if (!identical(arg07, _hashEnd)) {
-            result = _Jenkins.combine(result, arg07);
-            if (!identical(arg08, _hashEnd)) {
-              result = _Jenkins.combine(result, arg08);
-              if (!identical(arg09, _hashEnd)) {
-                result = _Jenkins.combine(result, arg09);
-                if (!identical(arg10, _hashEnd)) {
-                  result = _Jenkins.combine(result, arg10);
-                  if (!identical(arg11, _hashEnd)) {
-                    result = _Jenkins.combine(result, arg11);
-                    if (!identical(arg12, _hashEnd)) {
-                      result = _Jenkins.combine(result, arg12);
-                      if (!identical(arg13, _hashEnd)) {
-                        result = _Jenkins.combine(result, arg13);
-                        if (!identical(arg14, _hashEnd)) {
-                          result = _Jenkins.combine(result, arg14);
-                          if (!identical(arg15, _hashEnd)) {
-                            result = _Jenkins.combine(result, arg15);
-                            if (!identical(arg16, _hashEnd)) {
-                              result = _Jenkins.combine(result, arg16);
-                              if (!identical(arg17, _hashEnd)) {
-                                result = _Jenkins.combine(result, arg17);
-                                if (!identical(arg18, _hashEnd)) {
-                                  result = _Jenkins.combine(result, arg18);
-                                  if (!identical(arg19, _hashEnd)) {
-                                    result = _Jenkins.combine(result, arg19);
-                                    if (!identical(arg20, _hashEnd)) {
-                                      result = _Jenkins.combine(result, arg20);
-                                      // I can see my house from here!
-                                    }
-                                  }
-                                }
-                              }
-                            }
-                          }
-                        }
-                      }
-                    }
-                  }
-                }
-              }
-            }
-          }
-        }
-      }
-    }
-  }
-  return _Jenkins.finish(result);
-}
+  Object? arg01, Object? arg02, [
+  Object? arg03, Object? arg04,
+  Object? arg05, Object? arg06,
+  Object? arg07, Object? arg08,
+  Object? arg09, Object? arg10,
+  Object? arg11, Object? arg12,
+  Object? arg13, Object? arg14,
+  Object? arg15, Object? arg16,
+  Object? arg17, Object? arg18,
+  Object? arg19, Object? arg20, ]
+  ) => Object.hash(
```

Comment:
This forwarding will not work as expected.
Consider
```dart
hashValues(1, 2)
```
The original `hashValues` will call `1.hashCode` and `2.hashCode` do some combining operations.

The new version will call
```dart
Object.hash(1, 2, null, null, null, ... ,null)
```
`Object.hash` will 'see' that it is passed 20 values and call `.hashCode` on all of them and combine them.
This obviously will be slower (20 vs 2 calls to `.hashCode`).

The old `hashValues` and `Object.hash` both use the same trick - a private default sentinel - to detect the number of arguments. Since you can't access the private sentinel used by `Object.hash`, you can't forward arguments in a way that makes it behave 'the same'.

Recommendation:
Change _nothing_ in this file except for adding `@Deprecated` annotations.
If you don't change the code, you can't accidentally make some app worse.

Reply:
Agree with this!

Reply:
Thank you for the review! Makes perfect sense.

Looking back, having little deprecation knowledge, I just took inspiration from a few deprecated members and noticed that they were proxied to their replacements. Should have dug a little deeper.

Nevertheless, will fix this ASAP.

Reply:
If you *can* cleanly move to the new thing, it's generally a good thing. This is a gnarly case, though.

Reply:
Done!